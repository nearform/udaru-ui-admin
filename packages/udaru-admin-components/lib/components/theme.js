// Adapted from default theme on https://github.com/jxnblk/styled-system

// breakpoint values
// any array length works with styled-system
const breakpoints = ['500px', '800px', '1200px']

const colors = {
  primary: '#337ab7',
  primaryActive: '#02A171',
  danger: '#d9534f',
  lightGrey: '#D8D8D8',
  lightGrey2: '#888',
  lightGrey3: '#eee',
  darkGrey: '#555',
  black: '#333',
  green: '#5cb85c',
  lightBlue: '#5bc0de',
  orange: '#f0ad4e'
}

export { colors }

export const radius = '4px'

export const maxContainerWidth = '1280px'

// boxShadows
export const boxShadows = [
  `0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)`
]

export const speeds = {
  slow: '1s',
  medium: '.4s',
  fast: '.2s'
}

export const easings = {
  easeOut: 'cubic-bezier(0,.6,.3,1)',
  expoEaseOut: 'cubic-bezier(0,1,.3,1)',
  bounce: 'cubic-bezier(0,1.2,.5,1.4)'
}

// space is used for margin and padding scales
// it's recommended to use powers of two to ensure alignment
// when used in nested elements
// numbers are converted to px
const space = [0, 4, 8, 16, 32, 64, 128, 256, 512]

// typographic scale
const fontSizes = [12, 14, 16, 20, 24, 32, 48, 64, 96, 128]

// for any scale, either array or objects will work
const lineHeights = [1, 1.125, 1.25, 1.5]

const fontWeights = {
  normal: 500,
  bold: 700
}

const letterSpacings = {
  normal: 'normal',
  caps: '0.25em'
}

// border-radius
const radii = [0, 2, 4, 8]

const borders = [0, '1px solid', '2px solid']

const shadows = [`0 1px 2px 0 ${colors.text}`, `0 1px 4px 0 ${colors.text}`]

const theme = {
  breakpoints,
  colors,
  space,
  fontSizes,
  lineHeights,
  fontWeights,
  letterSpacings,
  radii,
  radius,
  borders,
  shadows,
  boxShadows,
  speeds,
  easings
}

export default theme
