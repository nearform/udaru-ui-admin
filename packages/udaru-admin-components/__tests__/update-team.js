import React from 'react'
import renderer from 'react-test-renderer'
import UpdateTeam from 'update-team'
import { createSerializer } from 'jest-emotion'
import * as emotion from 'emotion'
import theme from '../lib/components/theme'

expect.addSnapshotSerializer(createSerializer(emotion))

it('should render with default props', () => {
  const component = renderer.create(
    <UpdateTeam theme={theme} team={{ id: '1' }} />
  )
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

it('should warn user if no onCancel function is passed into component', () => {
  const component = renderer.create(<UpdateTeam theme={theme} />)
  const instance = component.root.instance
  const spy = jest.spyOn(global.console, 'log')

  expect(typeof instance.props.onCancel).toBe('function')
  instance.props.onCancel()

  expect(spy).toHaveBeenCalled()

  expect(global.console.log.mock.calls[0][0]).toBe(
    'WARNING: No onCancel function passed into the <UpdateTeam /> component.'
  )

  spy.mockReset()
  spy.mockRestore()
})

it('should reset state on dismissable error', () => {
  const component = renderer.create(
    <UpdateTeam theme={theme} team={{ id: '1' }} />
  )
  const instance = component.root.instance

  instance.setState({ success: true, hasError: true, errorMessage: 'foo' })
  expect(instance.state.success).toBeTruthy()

  instance.onDismiss()

  const { success, hasError, errorMessage } = instance.state

  expect(success).toBeFalsy()
  expect(hasError).toBeFalsy()
  expect(errorMessage).toBe('')
})

it('empty render if no team passed', () => {
  const component = renderer.create(<UpdateTeam theme={theme} />)
  const json = component.toJSON()

  expect(json).toMatchSnapshot()
})

it('should throw error in render and catch in error boundary', () => {
  const error = new Error('oops')
  const info = 'some info'
  const logError = jest.fn()
  const component = renderer.create(
    <UpdateTeam theme={theme} logError={logError} />
  )
  const instance = component.root.instance

  instance.componentDidCatch(error, info)

  expect(instance.state.hasError).toBeTruthy()
  expect(instance.state.errorMessage).toBe('An error has occured.')
  expect(logError).toHaveBeenCalled()
  expect(logError).toHaveBeenCalledWith(error, info)
})

it('should successfully submit form', async () => {
  const teamData = {
    id: '1',
    name: 'Team 1',
    description: 'Team description'
  }
  const teamToUpdate = {
    id: '1',
    name: 'Team 1 Updated',
    description: 'Team Description Update'
  }
  global.fetch = jest.fn().mockImplementation(
    () =>
      new Promise((resolve, reject) => {
        resolve({
          ok: true,
          json: function() {
            return teamData
          }
        })
      })
  )

  const component = renderer.create(
    <UpdateTeam
      theme={theme}
      udaruUrl="my-udaru-url"
      authorization="my-authorization"
      org="my-org"
      team={teamData}
    />
  )
  const instance = component.root.instance
  expect(instance.state.success).toBeFalsy()

  await instance.onFormSubmit(teamToUpdate)

  const {
    success,
    nonDismissableError,
    hasError,
    errorMessage
  } = instance.state
  expect(success).toBeTruthy()
  expect(hasError).toBeFalsy()
  expect(nonDismissableError).toBeFalsy()
  expect(errorMessage).toBe('')

  global.fetch.mockRestore()
})

it('should error if no url set', async () => {
  const component = renderer.create(
    <UpdateTeam
      theme={theme}
      authorization="my-authorization"
      org="my-org"
      team={{ id: '1' }}
    />
  )
  const instance = component.root.instance

  expect(instance.state.success).toBeFalsy()

  await instance.onFormSubmit({ name: '', description: '' })

  const {
    success,
    nonDismissableError,
    hasError,
    errorMessage
  } = instance.state

  expect(success).toBeFalsy()
  expect(hasError).toBeTruthy()
  expect(nonDismissableError).toBeFalsy()
  expect(errorMessage).toBe('Udaru URL prop not passed into component.')
})

it('should error if no authorization set', async () => {
  const component = renderer.create(
    <UpdateTeam
      theme={theme}
      udaruUrl="my-udaru-url"
      org="my-org"
      team={{ id: '1' }}
    />
  )
  const instance = component.root.instance

  expect(instance.state.success).toBeFalsy()

  await instance.onFormSubmit({ name: '', description: '' })

  const {
    success,
    nonDismissableError,
    hasError,
    errorMessage
  } = instance.state

  expect(success).toBeFalsy()
  expect(hasError).toBeTruthy()
  expect(nonDismissableError).toBeFalsy()
  expect(errorMessage).toBe('Authorization prop not passed into component.')
})

it('should handle error when updating team', async () => {
  const teamData = {
    id: '1',
    name: 'Team 1',
    description: 'Team description'
  }
  const teamToUpdate = {
    id: '1',
    name: 'Team 1 Updated',
    description: 'Team Description Update'
  }
  global.fetch = jest.fn().mockImplementation(
    () =>
      new Promise((resolve, reject) => {
        resolve({
          ok: false,
          json: function() {
            return teamData
          }
        })
      })
  )

  const component = renderer.create(
    <UpdateTeam
      theme={theme}
      udaruUrl="my-udaru-url"
      authorization="my-authorization"
      org="my-org"
      team={teamData}
    />
  )
  const instance = component.root.instance
  expect(instance.state.success).toBeFalsy()

  await instance.onFormSubmit(teamToUpdate)

  const {
    success,
    nonDismissableError,
    hasError,
    errorMessage
  } = instance.state

  expect(success).toBeFalsy()
  expect(hasError).toBeTruthy()
  expect(nonDismissableError).toBeFalsy()
  expect(errorMessage).toBe('error saving that team.')

  global.fetch.mockRestore()
})
