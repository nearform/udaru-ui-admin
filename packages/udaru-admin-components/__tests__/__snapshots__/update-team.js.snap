// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty render if no team passed 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="alert alert-danger"
        role="alert"
      >
        <strong>
          Oops! There was an error
        </strong>
        <p>
          No Team ID passed into component.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render with default props 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="page-header"
      >
        <h1>
          Update Team
        </h1>
      </div>
    </div>
    <div
      className="col-xs-12"
    >
      <div
        className="alert alert-danger"
        role="alert"
      >
        All fields in 
        <strong>
          RED
        </strong>
         are required.
      </div>
    </div>
    <div
      className="col-xs-12"
    >
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
            htmlFor={undefined}
          >
            ID
          </label>
          <input
            className="form-control"
            disabled={true}
            id={undefined}
            placeholder="ID"
            type="text"
            value="1"
          />
        </div>
        <div
          className="form-group has-feedback has-error"
        >
          <label
            className="control-label"
            htmlFor="name"
          >
            Name
          </label>
          <input
            className="form-control"
            id="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
          />
          <span
            className="form-control-feedback glyphicon glyphicon-remove"
          />
        </div>
        <div
          className="form-group has-feedback has-error"
        >
          <label
            className="control-label"
            htmlFor="description"
          >
            Description
          </label>
          <input
            className="form-control"
            id="description"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Description"
            type="text"
            value=""
          />
          <span
            className="form-control-feedback glyphicon glyphicon-remove"
          />
        </div>
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Submit
        </button>
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </form>
    </div>
  </div>
</div>
`;
